src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -58,24 +53,22 @@ public TransactionActivateHandler(
             PaymentRequestsInfoRepository paymentRequestsInfoRepository,
             TransactionsEventStoreRepository<TransactionActivatedData> transactionEventActivatedStoreRepository,
             TransactionsEventStoreRepository<TransactionActivationRequestedData> transactionEventActivationRequestedStoreRepository,
-            EcommerceSessionsClient ecommerceSessionsClient,
             NodoOperations nodoOperations,
             @Qualifier("transactionActivatedQueueAsyncClient") QueueAsyncClient transactionActivatedQueueAsyncClient,
             @Value("${payment.token.validity}") Integer paymentTokenTimeout
     ) {
         this.paymentRequestsInfoRepository = paymentRequestsInfoRepository;
         this.transactionEventActivatedStoreRepository = transactionEventActivatedStoreRepository;
         this.transactionEventActivationRequestedStoreRepository = transactionEventActivationRequestedStoreRepository;
-        this.ecommerceSessionsClient = ecommerceSessionsClient;
         this.nodoOperations = nodoOperations;
         this.paymentTokenTimeout = paymentTokenTimeout;
         this.transactionActivatedQueueAsyncClient = transactionActivatedQueueAsyncClient;
     }
 
-    public Mono<Tuple3<Mono<TransactionActivatedEvent>, Mono<TransactionActivationRequestedEvent>, SessionDataDto>> handle(
-                                                                                                                           TransactionActivateCommand command
+    public Mono<Tuple2<Mono<TransactionActivatedEvent>, Mono<TransactionActivationRequestedEvent>>> handle(
+                                                                                                           TransactionActivateCommand command
     ) {
-        final String transactionId = UUID.randomUUID().toString();
```

Comment:
the `TransactionActivateCommand` doesn't have the `transactionId` which is generated here, why delete it?

Reply:
The transaction id is now generated into the TransactionController just because it's needed for JWT token generation. I'm going to move it back here since token will be generated in place of session service call

Reply:
Resolved into https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/121/commits/f1619dca36997c33067c1f7b842e05e64942c277